controllers/openstackconfiggenerator.go

```diff
@@ -121,7 +121,10 @@ func (r *OpenStackConfigGeneratorReconciler) Reconcile(ctx context.Context, req
 	//
 	// initialize condition
 	//
-	cond := &ospdirectorv1beta1.Condition{}
+	cond := instance.Status.Conditions.GetCurrentCondition()
+	if cond == nil {
+		cond = &ospdirectorv1beta1.Condition{}
+	}
```

Comment:
could then use in all those just instance.Status.Conditions.InitCondition()

Reply:
Ack